### PR TITLE
implemented timer procedure

### PR DESCRIPTION
**Pull request by SergeyPjatkin**
added timer procedure

### CODE REVIEW

File: app/src/main/java/com/univer/covid19help/MainFragment.kt [hunk 1]
```diff
@@ -202,6 +222,118 @@ class MainFragment : Fragment() {
...
+
+    private fun stopTimer() {
+        Log.i(TAG, "stopTimer")
+        isTimerRunning = false
+        resetTimeDiff()
+    }
+
+    private fun onTimer() {
+        when(state) {
+            VaccineState.SelectFirstDate -> {
```
Comment (elivina997) [inline app/src/main/java/com/univer/covid19help/MainFragment.kt:247]: Consider to rename SelectFirstDate to Idle

File: app/src/main/java/com/univer/covid19help/MainFragment.kt [hunk 2]
```diff
@@ -202,6 +222,118 @@ class MainFragment : Fragment() {
...
+    private fun resetTimeDiff() {
+        alreadyLessThanHour = false
+        alreadyLessThan30Min = false
+        alreadyLessThan15Min = false
+        alreadyLessThan5Min = false
+        alreadyWasNow = false
+    }
+
+    private fun getTimeDiff(current:Date, future: Date): TimeDiff {
+        val HOUR: Long = 1000*60*60
```
Comment (elivina997) [inline app/src/main/java/com/univer/covid19help/MainFragment.kt:298]: Consider to use global class variable

Comment (SergeyPjatkin) [reply app/src/main/java/com/univer/covid19help/MainFragment.kt:298]: Done

File: app/src/main/java/com/univer/covid19help/MainFragment.kt (hunk 1 above)
Comment (SergeyPjatkin) [reply app/src/main/java/com/univer/covid19help/MainFragment.kt:247]: done

File: app/src/main/java/com/univer/covid19help/MainFragment.kt (hunk 2 above)
Comment (elivina997) [reply app/src/main/java/com/univer/covid19help/MainFragment.kt]: Ok

**Review by elivina997 — APPROVED** [review]

**Pull request merged**